### ISSUE TITLE
Dependency generator for shell

### ISSUE THREAD

**Issue by ignatenkobrain** [rpm-software-management/rpm#628]
It seems that b780f5a8d636073d67dbcdfa3f8795f878b55218 (from 2007) removed such generator as not being used which probably happened because it was not ported to new dependency generators syntax or so.

It would be nice if we could generate dependencies on scriptlets using that code. No more Requires(post): /sbin/install-info!

**Comment by pmatilai** [comment]
Um, what? The stuff removed in that commit had never been used at all in rpm, and scriptlet dependency extraction has never been implemented, much less removed.

I know about bash --rpm-requires, but the results you get are so far off that it's more trouble than actually helping anything.

**Comment by ignatenkobrain** [comment]
@pmatilai for the simple scriptlets which call ldconfig or anything else simple (few commands), I think using `bash --rpm-requires` would be more than enough and will be more nice to users so they don't have to write `Requires(post): /usr/sbin/semanage` or anything like that.

**Comment by pmatilai** [comment]
Look, I've poked at it several times over the years. Unless there's been some major advances in that department, it's just more trouble than its worth.

**Comment by pmatilai** [comment]
Mind you, the above doesn't mean I wouldn't consider a nice patch adding support for scriptlet dependency autogeneration: from bash --rpm-requires output, at least "executable(/absolute/path)" entries should be more-or-less safely usable after just mangling them to regular file dependencies. It's just that it doesn't seem worth implementing to myself.

**Comment by ignatenkobrain** [comment]
I never asked you to implement it :)

I try to always open an issue and discuss whether it is something which we would get in before doing the actual implementation.

**Comment by jasontibbitts** [comment]
I do think this could be useful if it would serve to reduce packager workload, though the trend seems to be moving away from scriptlets in general.

**Comment by pmatilai** [comment]
I had yet another look at this and it seems just as unfeasible as it always did:
- The vast majority of commands in scriptlets lack absolute paths, so it emits executable(fu), which would need packages to provide the basenames of all their executables to match these, which can cause wrong packages to be required when the path is stripped off. The alternative is to map these to paths at build-time, but the scriptlet commands aren't necessarily build requires so we couldnt' know what the right path is, unless we ask packagers to add them as build-requires at which point .. what was the point again? Oh right, reduce the dependencies packagers need to add manually!
- if we limited the rpm-used dependencies to absolute paths, then the experience will be wildly inconsistent, which I think is even worse for packager sanity
- even if we ignored the inconsistency, `bash --rpm-requires` introduces dependencies for constructs that specifically try to avoid that dependency, eg this is picked up as a dependency on /usr/bin/fubar:
```
if [ -x /usr/bin/fubar ] ; then
        /usr/bin/fubar --something
fi
```

So, no. This isn't worth the trouble and all the false positives and whatnot.
